src/main/java/thePackmaster/cardmodifiers/artificerpack/BlockEnchantModifier.java

```diff
@@ -25,13 +25,14 @@ public BlockEnchantModifier(int amount) {
     @Override
     public void onInitialApplication(AbstractCard card) {
         for (AbstractCardModifier m : CardModifierManager.modifiers(card)) {
-            if (m instanceof BlockEnchantModifier) {
-                if (((BlockEnchantModifier) m).amount != 0) {
-                    ((BlockEnchantModifier) m).amount += amount;
-                    amount = 0;
-                }
+            if (m instanceof BlockEnchantModifier && m != this) {
+                ((BlockEnchantModifier) m).amount += amount;
+                amount = 0;
             }
         }
+        if (amount == 0) {
+            CardModifierManager.removeSpecificModifier(card, this, true);
```

Comment:
Might want to make this an anonymous action. This could lead to a conc mod exception